### ISSUE TITLE
Change SQLServerDataType.TIMESTAMP to be of type VARBINARY / byte[]

### ISSUE THREAD

**Issue by Kaleidoscoper** [jOOQ/jOOQ#7341]
### Expected behavior and actual behavior:
Timestamp is special in sqlserver, it is a binary stream describing row version.While when import the data containing the column, getLong() in DefaultResultSet.java is called resulting in a SQLServerException: The conversion from timestamp to BIGINT is unsupported.
### Steps to reproduce the problem:
Create a table A including a timestamp column C in sqlserver 2012 ,and create a query like "select */C from A" and calling fetch the SQLServerException shows up.
### Versions:

- jOOQ:3.10.2
- Java:jdk 1.8.0_144
- Database (include vendor): sqlserver 2012 
- OS:Mac
- JDBC Driver (include name if inofficial driver):mssql-jdbc 6.1.0.jre8

**Comment by lukaseder** [comment]
Thank you very much for your report and sorry for the delay. Will look into this shortly

**Comment by lukaseder** [comment]
Regarding the reproduction of this issue, could you quickly show a DDL example of such a table, and the actual jOOQ code? (i.e. are you using generated code, or a plain SQL query)?

**Comment by Kaleidoscoper** [comment]
Here is my schema
create TABLE student
( id int,
  name VARCHAR(255),
  timestamp TIMESTAMP
);
Here is my insertion
INSERT into student(id,name) VALUES (1,'whj');
Here is my code
Connection conn = DriverManager.getConnection(url,userName,password);
DSLContext create = DSL.using(conn,SQLDialect.SQLSERVER);
Result<?> result = create.select().from(table("student")).fetch();
and I got an exception 
Caused by: com.microsoft.sqlserver.jdbc.SQLServerException: The conversion from timestamp to BIGINT is unsupported.
	at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDriverError(SQLServerException.java:206)
	at com.microsoft.sqlserver.jdbc.DataTypes.throwConversionError(DataTypes.java:1652)
	at com.microsoft.sqlserver.jdbc.ServerDTVImpl.getValue(dtv.java:4017)
	at com.microsoft.sqlserver.jdbc.DTV.getValue(dtv.java:237)
	at com.microsoft.sqlserver.jdbc.Column.getValue(Column.java:162)
	at com.microsoft.sqlserver.jdbc.SQLServerResultSet.getValue(SQLServerResultSet.java:2100)
	at com.microsoft.sqlserver.jdbc.SQLServerResultSet.getValue(SQLServerResultSet.java:2085)
	at com.microsoft.sqlserver.jdbc.SQLServerResultSet.getLong(SQLServerResultSet.java:2346)
	at org.jooq.tools.jdbc.DefaultResultSet.getLong(DefaultResultSet.java:139)
	at org.jooq.impl.CursorImpl$CursorResultSet.getLong(CursorImpl.java:831)
	at org.jooq.impl.DefaultBinding.get(DefaultBinding.java:1529)
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.setValue(CursorImpl.java:1714)
       …..
The timestamp type in sqlserver is actually a binary stream for record version, so i think if it’s improper to fetch it using getLong?
Thks for replying :)

**Comment by lukaseder** [comment]
Thank you very much for detailing this. I see, indeed, we have to fetch a `byte[]` type for the `TIMESTAMP` type. In order to fix this on your side, you will need to specify each column explicitly, or not select your timestamp column. E.g. this should work:

```java
Result<?> result = create
    .select(
        field("id", Integer.class),
        field("name", String.class),
        field("timestamp", byte[].class))
    .from(table("student"))
    .fetch();
```

In the meantime, we'll change `SQLServerDataType.TIMESTAMP` to be a synonym for the `SQLDataType.VARBINARY` type.

**Comment by lukaseder** [comment]
Fixed this for jOOQ 3.11

**Comment by Kaleidoscoper** [comment]
Thank you for your replying, the additional byte[].class does make it work, while I really hope the jooq will automatically finish the type conversion:)

**Comment by lukaseder** [comment]
@Kaleidoscoper What do you mean by this?

> while I really hope the jooq will automatically finish the type conversion:)

**Comment by Kaleidoscoper** [comment]
I mean it would be better if jooq can automatically convert timestamp type of sqlserver to a varbinary type instead of explicitly define a byte[].class. Cuz sometimes i don’t know the column’s type but the column’s name.

**Comment by lukaseder** [comment]
I'm sorry I have trouble understanding what you mean. The fix changed the definition of the existing `TIMESTAMP` data type configuration from `DataType<Long>` to `DataType<byte[]>`. This means that you don't have to do anything to get "varbinary" or "byte[]" behaviour from jOOQ 3.11 onwards.

Is there anything else you'd like jOOQ to do? If so, would you mind providing an example?

**Comment by Kaleidoscoper** [comment]
Sorry for misleading you,I mean your fix in 3.11 could really help me and I'm with great expectation for that and how soon will 3.11 be published?
